### PR TITLE
Cloudsync: dropbox token re-generation

### PR DESCRIPTION
**Pull request by weijiuqiao**
This pr adds support for dropbox short-lived token for the cloudsync module. See #9861

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9862)

### CODE REVIEW

**Comment by Frenzie** [comment]
```
frontend/apps/cloudstorage/syncservice.lua:92:12: accessing undefined variable 'sever'
```

File: frontend/apps/cloudstorage/syncservice.lua [hunk 1]
```diff
@@ -87,7 +87,11 @@ function SyncService.getReadablePath(server)
     local url = util.stringStartsWith(server.url, "/") and server.url:sub(2) or server.url
     url = util.urlDecode(url) or url
     url = util.stringEndsWith(url, "/") and url or url .. "/"
-    url = (server.address:sub(-1) == "/" and server.address or server.address .. "/") .. url
+    if server.type == "dropbox" then
+        url = "/" .. url
+    elseif sever.type == "webdav" then
```
Comment (Frenzie) [inline frontend/apps/cloudstorage/syncservice.lua:92]:
```suggestion
    elseif server.type == "webdav" then
```

File: frontend/apps/cloudstorage/syncservice.lua [hunk 2]
```diff
@@ -132,11 +136,15 @@ function SyncService.sync(server, file_path, sync_cb, is_silent)
     local code_response = 412 -- If-Match header failed
     local etag
     local api = server.type == "dropbox" and require("apps/cloudstorage/dropboxapi") or require("apps/cloudstorage/webdavapi")
+    local token = server.password
+    if server.type == "dropbox" then
+        token = api:getAccessToken(server.password, server.address) or server.password -- in case using long-term token
```
Comment (hius07) [inline frontend/apps/cloudstorage/syncservice.lua:141]: To understand that we use the long-lived token we can just check:
`self.address == nil or self.address == ""`
and not to ask Dropbox and get the error.
https://github.com/koreader/koreader/blob/fa9f0acf5e58dcc4f68d4effbdaefa795c6b67c4/frontend/apps/cloudstorage/cloudstorage.lua#L135-L148

**Comment by hius07** [comment]
https://github.com/koreader/koreader/blob/40d5e1654c1f240a181fa4da95fab321ea706a57/plugins/statistics.koplugin/main.lua#L1168-L1170
Usually if wi-fi is off and a user asks for a service that requires connection, the dialog "Turn on wi-fi?" pops up.
https://github.com/koreader/koreader/blob/fa9f0acf5e58dcc4f68d4effbdaefa795c6b67c4/frontend/apps/cloudstorage/cloudstorage.lua#L153-L155

File: frontend/apps/cloudstorage/syncservice.lua (hunk 2 above)
Comment (weijiuqiao) [reply frontend/apps/cloudstorage/syncservice.lua]: Thanks for pointing this out. Done now.

**Comment by hius07** [comment]
Thank you, looks good.

Please note that in the settings dialog SyncService.getReadablePath shows `server.address` (which for short-lived tokens contains app_key:app_secret).

**Comment by weijiuqiao** [comment]
>Please note that in the settings dialog SyncService.getReadablePath shows server.address

It’s been fixed in this pr :)

**Pull request merged**